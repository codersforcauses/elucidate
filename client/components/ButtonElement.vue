<template>
  <button><slot></slot></button>
</template>

<script>
export default {
  name: 'ButtonElement',
};
</script>

<style></style>
