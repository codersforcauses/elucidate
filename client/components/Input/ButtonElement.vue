<template>
  <nuxt-link v-if="to !== undefined" :to="to">
    <button
      v-if="type === 'primary'"
      :type="submitType"
      :class="`bg-white border-solid text-pink-600 border-pink-600 border-2 ${commonBtnStyles}`"
    >
      <slot></slot>
    </button>

    <button
      v-else
      :class="`bg-white text-primary border border-solid border-primary ${commonBtnStyles}`"
    >
      <slot></slot>
    </button>
  </nuxt-link>
  <span v-else>
    <button
      v-if="type === 'primary'"
      :class="`bg-white border-solid text-pink-600 border-pink-600 border-2 ${commonBtnStyles}`"
    >
      <slot></slot>
    </button>

    <button
      v-else
      :class="`bg-white text-primary border border-solid border-primary ${commonBtnStyles}`"
    >
      <slot></slot>
    </button>
  </span>
</template>

<script>
export default {
  name: 'ButtonElement',
  props: {
    submitType: {
      type: String,
      default: undefined,
    },
    to: {
      type: String,
      default: undefined,
    },
    type: {
      type: String,
      default: 'primary',
    },
  },
  data: () => ({
    commonBtnStyles:
      'h-10 w-40 rounded-md hover:shadow-lg transition-shadow duration-200 ease-in-out content-center',
  }),
};
</script>
