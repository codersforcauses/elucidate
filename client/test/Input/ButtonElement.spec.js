import ButtonElement from '@/components/Input/ButtonElement.vue';
import { mount } from '@vue/test-utils';

describe('ButtonElement', () => {
  it('is a Vue instance', () => {
    const wrapper = mount(ButtonElement);
    expect(wrapper.vm).toBeTruthy();
  });

  it('renders correctly', () => {
    const wrapper = mount(ButtonElement);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays text', () => {
    const wrapper = mount(ButtonElement, {
      slots: {
        default: 'Test',
      },
    });
    expect(wrapper.find('button').text()).toBe('Test');
  });
});
