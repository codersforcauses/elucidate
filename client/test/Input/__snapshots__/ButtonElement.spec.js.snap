// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonElement renders correctly 1`] = `
<span>
  <button
    class="text-white bg-primary h-10 w-32 rounded-md hover:shadow-lg transition-shadow duration-200 ease-in-out"
  />
   
  <!---->
   
  <!---->
</span>
`;
